File: ai25btech11031/Assignments/Matgeo/2.10.19/Codes/solution.py
def dot(a, b):
    return a[0]*b[0] + a[1]*b[1] + a[2]*b[2]

def cross(a, b):
    return [
        a[1]*b[2] - a[2]*b[1],
        a[2]*b[0] - a[0]*b[2],
        a[0]*b[1] - a[1]*b[0]
    ]

# Input vectors
u = list(map(int, input("Enter vector u (x y z): ").split()))
v = list(map(int, input("Enter vector v (x y z): ").split()))
w = list(map(int, input("Enter vector w (x y z): ").split()))

# A = u · (v × w)
A = dot(u, cross(v, w))

# B = v · (u × w)
B = dot(v, cross(u, w))

# C = (v × w) · u
C = dot(cross(v, w), u)

# D = (u × v) · w
D = dot(cross(u, v), w)

# Print results
print("\nResults:")
print(f"A = u · (v × w) = {A}")
print(f"B = v · (u × w) = {B}")
print(f"C = (v × w) · u = {C}")
print(f"D = (u × v) · w = {D}")

# Check which is different
if A == C == D and B != A:
    print("\n=> Expression B is different.")
elif B == C == D and A != B:
    print("\n=> Expression A is different.")
elif A == B == D and C != A:
    print("\n=> Expression C is different.")
elif A == B == C and D != A:
    print("\n=> Expression D is different.")
else:
    print("\n=> No unique difference found (all may be equal or zero).")

File: ai25btech11031/Assignments/Matgeo/2.10.19/Codes/Solution.c
#include <stdio.h>
#include "Solution.h"

int main() {
    int u[3], v[3], w[3];
    int temp[3];
    int A, B, C, D;

    // Input vectors
    printf("Enter vector u (x y z): ");
    scanf("%d %d %d", &u[0], &u[1], &u[2]);
    printf("Enter vector v (x y z): ");
    scanf("%d %d %d", &v[0], &v[1], &v[2]);
    printf("Enter vector w (x y z): ");
    scanf("%d %d %d", &w[0], &w[1], &w[2]);

    // A = u · (v × w)
    cross(v, w, temp);
    A = dot(u, temp);

    // B = v · (u × w)
    cross(u, w, temp);
    B = dot(v, temp);

    // C = (v × w) · u
    cross(v, w, temp);
    C = dot(temp, u);

    // D = (u × v) · w
    cross(u, v, temp);
    D = dot(temp, w);

    // Print results
    printf("\nResults:\n");
    printf("A = u · (v × w) = %d\n", A);
    printf("B = v · (u × w) = %d\n", B);
    printf("C = (v × w) · u = %d\n", C);
    printf("D = (u × v) · w = %d\n", D);

    // Check which is different
    if (A == C && A == D && B != A) {
        printf("\n=> Expression B is different.\n");
    } else if (B == C && B == D && A != B) {
        printf("\n=> Expression A is different.\n");
    } else if (A == B && A == D && C != A) {
        printf("\n=> Expression C is different.\n");
    } else if (A == B && A == C && D != A) {
        printf("\n=> Expression D is different.\n");
    } else {
        printf("\n=> No unique difference found (all may be equal or zero).\n");
    }

    return 0;
}

